### ISSUE TITLE
agent卸载后加载报异常

### ISSUE THREAD

**Issue by zhzx2162626** [sermant-io/Sermant#1518]
### What do you want to ask?

插件通过attach方式加载后，通过attach的方式先卸载，再进行加载会报异常DupServiceException。

**Comment by hanbingleixue** [comment]
@zhzx2162626  Which plugin did you install and uninstall?  Please provide the specific error logs for us to locate the proble. In addition, use the bug report template to submit issues. Describe the problem clearly.

**Comment by zhzx2162626** [comment]
代码片段
...
private String install = "command=INSTALL-PLUGINS:qoc-uni-plugin";
private String uninstall = "command=UNINSTALL-PLUGINS:qoc-uni-plugin";

public void exec(string cmd){
....
VirtualMachine vm = VirtualMachine.attach(pid);
vm.loadAgent("/myagent.jar", cmd);
vm.detach();
}

public void reload(){
exec("uninstall");
exec("install");
}
--------------

调用 reload之后发生如下异常
![WechatIMG86](https://github.com/sermant-io/Sermant/assets/15646108/48d695bb-052c-40c3-80be-0b6f57e1abf2)

**Comment by hanbingleixue** [comment]
@zhzx2162626 The possible cause is that the LocalMockHttpService is not properly stopped during the uninstallation. As a result, this problem occurs when the plug-in is installed for the second time. You can check whether the logs generated during the uninstallation contain any exception information. Alternatively, run the debug command to check whether LocalMockHttpService is properly stopped during uninstallation and whether LocalMockHttpService is removed from ServiceManager after being stopped.

**Comment by zhzx2162626** [comment]
经过debug我感觉发现问题了，是因为在卸载插件的时候只有plugin里的服务的实现类被卸载了，但是接口类没有卸载；在重新安装插件的时候因为接口类仍然在service的map里导致冲突，不知道这是否一个bug；
然后我解决的方法是卸载的时候直接卸载agent而不是卸载插件，然后再重新安装agent，不知道是否可以？

**Comment by hanbingleixue** [comment]
@zhzx2162626 You can directly uninstall the Agent. After the Agent is uninstalled, the map of the service is cleared. 
Another thing I want to ask：
In which module is your interface class defined?  Is it in the plugin module?  What class loader do you use to load your interface classes?

**Comment by zhzx2162626** [comment]
![WechatIMG87](https://github.com/sermant-io/Sermant/assets/15646108/9712f60d-ba62-48de-8abf-0f72ed55487d)
这是我们的l目录结构，不知道能不能够提供信息
此外
<img width="709" alt="image" src="https://github.com/sermant-io/Sermant/assets/15646108/c4887b34-98ad-41c8-9df4-15ad48bafd1b">
我们用的版本是1.2.0，我看下一个版本刚好有个bugfix，但是看不到这个是修复了什么

**Comment by hanbingleixue** [comment]
@zhzx2162626 The next version fixes the problem that PluginClassLoader can be loaded into the host class, which is not the same as your problem. But you can try with the latest version if you still have the problem

**Comment by lilai23** [comment]
I have reproduce it using 1.2.0 and it has been fixed in 1.2.1 and newer version. So you can update your sermant version
